app/services/schedules/find.rb

```diff
@@ -0,0 +1,94 @@
+module Schedules
+  class Find
+    attr_accessor :period, :training_programme, :started_on
+
+    def initialize(period:, training_programme:, started_on:)
+      @period = period
+      @training_programme = training_programme
+      @started_on = started_on
+    end
+
+    def call
+      return unless provider_led?
+      return most_recent_schedule if previous_provider_led_periods.exists?
+
+      Schedule.find_by(contract_period_year:, identifier:)
```

Comment:
Minor improv - I would make this a private method so I can just easily read what is happening in the call e.g. `find_standard_schedule`